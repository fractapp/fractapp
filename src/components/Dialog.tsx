import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {BlueButton} from 'components/BlueButton';
import PopDialog, {DialogContent} from 'react-native-popup-dialog';

/**
 * Popup dialog component with 1 button
 * @category Components
 */
export const Dialog = ({
  title,
  text,
  visible,
  onPress,
}: {
  title: string;
  text: string;
  visible: boolean;
  onPress: () => void;
}) => {
  return (
    <PopDialog
      visible={visible}
      width={0.7}
      dialogTitle={<Text style={styles.title}>{title}</Text>}
      footer={
        <View style={styles.btnImg}>
          <BlueButton text={'OK'} height={40} onPress={onPress} />
        </View>
      }>
      <DialogContent style={styles.dialog}>
        <Text style={styles.description}>{text}</Text>
      </DialogContent>
    </PopDialog>
  );
};

const styles = StyleSheet.create({
  dialog: {
    alignItems: 'center',
    padding: 0,
  },
  title: {
    marginTop: 10,
    textAlign: 'center',
    fontSize: 17,
    fontFamily: 'Roboto-Regular',
    color: 'black',
  },
  description: {
    marginTop: 5,
    width: '100%',
    textAlign: 'center',
    fontSize: 14,
    fontFamily: 'Roboto-Regular',
    color: '#888888',
  },
  btnImg: {
    marginBottom: 12,
    width: '40%',
    alignSelf: 'center',
  },
});
