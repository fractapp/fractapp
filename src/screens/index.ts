export {ImportWallet} from './ImportWallet';
export {SaveSeed} from './SaveSeed';
export {SaveWallet} from './SaveWallet';
export {SettingWallet} from './SettingWallet';
export {Start} from './Start';
export {WalletDetails} from './WalletDetails';
export {WalletFileBackup} from './WalletFileBackup';
export {Wallets} from './Wallets';
export {Settings} from './Settings';
export {ConfirmSaveSeed} from './ConfirmSaveSeed';
export {ImportSeed} from './ImportSeed';
export {WalletFileImport} from './WalletFileImport';
export {GoogleDrivePicker} from './GoogleDrivePicker';
export {Receive} from './Receive';
export {Milestone} from './Milestone';
export {TransactionDetails} from './TransactionDetails';
export {SplashScreen} from './SplashScreen';
export {Backup} from './Backup';
export {NewPassCode} from './NewPassCode';
export {VerifyPassCode} from './VerifyPassCode';
export {EditProfile} from './EditProfile';
export {EditPhoneNumber} from './EditPhoneNumber';
export {SelectCountry} from './SelectCountry';
export {ConfirmCode} from './ConfirmCode';
export {EditEmail} from './EditEmail';
export {Navigation} from './Navigation';
export {Chats} from './Chats';
export {Chat} from './Chat';
export {Connecting} from './Connecting';
export {EditUsername} from './EditUsername';
export {EditName} from './EditName';
export {Search} from './Search';
export {SelectWallet} from './SelectWallet';
export {Send} from './Send';
export {EnterAmount} from './EnterAmount';
export {SendWithEnterAddress} from './SendWithEnterAddress';
