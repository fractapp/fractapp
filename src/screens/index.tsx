export { default as ImportWallet } from './ImportWallet';
export { default as SaveSeed } from './SaveSeed';

export { default as SaveWallet } from './SaveWallet';
export { default as SettingWallet } from './SettingWallet';
export { default as Start } from './Start';

export { default as WalletDetails } from './WalletDetails';

export { default as WalletEncryption } from './WalletEncryption';
export { default as Wallets } from './Wallets';
export { ConfirmSaveSeed } from './ConfirmSaveSeed'