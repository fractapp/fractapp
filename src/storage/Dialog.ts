import {createContext, Dispatch} from 'react';

/**
 * @namespace
 * @category Context storage
 */
namespace DialogStore {
  export enum Action {
    OPEN,
    CLOSE,
  }

  type State = {
    title: string;
    text: string;
    visible: boolean;
    onPress?: () => void;
  };

  export const initialState = (): State => ({
    text: '',
    title: '',
    visible: false,
  });

  export type ContextType = {
    state: State;
    dispatch: Dispatch<any>;
  };

  export const Context = createContext<ContextType>({
    state: initialState(),
    dispatch: () => null,
  });

  export function reducer(prevState: State, action: any): State {
    switch (action.type) {
      case Action.OPEN:
        return {
          title: action.title,
          text: action.text,
          onPress: action.onPress,
          visible: true,
        };
      case Action.CLOSE:
        let newDialog = Object.assign({}, prevState);
        newDialog.visible = false;
        return newDialog;
      default:
        return prevState;
    }
  }

  export const open = (title: string, text: string, onPress: () => void) => ({
    type: Action.OPEN,
    title: title,
    text: text,
    onPress: onPress,
  });
  export const close = () => ({
    type: Action.CLOSE,
  });
}
export default DialogStore;
