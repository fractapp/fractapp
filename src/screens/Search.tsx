import React, {useContext, useEffect, useState} from 'react';
import {
  FlatList,
  StyleSheet,
  TextInput,
  TouchableHighlight,
  View,
  PermissionsAndroid,
} from 'react-native';
import Contacts from 'react-native-contacts';
import {isValidPhoneNumber, parsePhoneNumber} from 'react-phone-number-input';
import Ionicons from 'react-native-vector-icons/Ionicons';
import {Contact} from 'components/Contact';
import {SendBy} from 'components/SendBy';
import backend from 'utils/backend';
import GlobalStore from 'storage/Global';
import Dialog from 'storage/Dialog';
import DialogStore from 'storage/Dialog';
import {UserProfile} from 'types/profile';
import {Wallet} from 'types/wallet';
import {ChatType} from 'types/chatInfo';

export const Search = ({navigation, route}: {navigation: any; route: any}) => {
  const wallet: Wallet = route.params?.wallet;

  const globalContext = useContext(GlobalStore.Context);
  const dialogContext = useContext(DialogStore.Context);
  const [searchString, setSearchString] = useState<string>('');
  const [users, setUsers] = useState<Array<UserProfile>>();

  useEffect(() => {
    getMyMatchContacts();
    if (!globalContext.state.isRegistered) {
      return;
    }

    PermissionsAndroid.request(
      PermissionsAndroid.PERMISSIONS.READ_CONTACTS,
    ).then(async (status) => {
      if (status === 'never_ask_again') {
        dialogContext.dispatch(
          Dialog.open(
            'Open settings',
            'If you want to find users by your contacts then open the application settings and give it access to read your contacts .',
            () => dialogContext.dispatch(Dialog.close()),
          ),
        );
        return;
      }
      if (status !== 'granted') {
        return;
      }

      const contacts = await Contacts.getAll();

      const backendContacts = await backend.myContacts();

      const mapBackendContacts = new Map<string, boolean>();
      for (let v of backendContacts) {
        mapBackendContacts.set(v, true);
      }
      const validContacts = new Map<string, boolean>();
      for (let contact of contacts) {
        if (contact.phoneNumbers.length === 0) {
          continue;
        }

        for (let c of contact.phoneNumbers) {
          if (!isValidPhoneNumber(c.number)) {
            continue;
          }
          const number = parsePhoneNumber(c.number)?.number;
          if (number === undefined) {
            continue;
          }
          if (mapBackendContacts.has(number)) {
            continue;
          }
          validContacts.set(number, true);
        }
      }

      let result = Array.from(validContacts, ([number]) => number);
      if (result.length > 0) {
        await backend.updateContacts(result);
        await getMyMatchContacts();
      }
    });
  }, []);

  const getMyMatchContacts = async () => {
    let contacts = await backend.myMatchContacts();

    contacts = contacts.filter(
      (user) => user.id !== globalContext.state.profile.id,
    );
    setUsers(contacts);
    const ids = new Array<string>();
    for (let user of contacts) {
      globalContext.dispatch(GlobalStore.setUser(user));
      ids.push(user.id);
    }

    globalContext.dispatch(GlobalStore.setContacts(ids));
  };

  useEffect(() => {
    if (searchString.length === 0) {
      const contacts = new Array<UserProfile>();
      for (let id of globalContext.state.contacts) {
        if (!globalContext.state.users.has(id)) {
          continue;
        }
        contacts.push(globalContext.state.users.get(id)!);
      }
      setUsers(contacts);
    } else {
      if (searchString[0] === '@') {
        backend.search(searchString.split('@')[1], false).then((users) => {
          setUsers(
            users.filter((user) => user.id !== globalContext.state.profile.id),
          );
        });
      } else {
        backend.search(searchString, true).then((users) => {
          setUsers(
            users.filter((user) => user.id !== globalContext.state.profile.id),
          );
        });
      }
    }
  }, [searchString]);

  const renderItem = ({item}: {item: any}) => {
    return (
      <TouchableHighlight
        onPress={() => {
          globalContext.dispatch(GlobalStore.setUser(item));

          navigation.navigate('Chat', {
            chatInfo: {
              id: item.id,
              name: item.name,
              lastTxId: '',
              lastTxCurrency: 0,
              notificationCount: 0,
              timestamp: 0,
              type: ChatType.Chat,
              details: null,
            },
          });
        }}
        underlayColor="#f8f9fb">
        <Contact
          name={item.name}
          img={
            item.avatarExt === ''
              ? require('assets/img/default-avatar.png')
              : {
                  uri: backend.getImgUrl(
                    item.id,
                    item.avatarExt,
                    item.lastUpdate,
                  ),
                }
          }
          usernameOrPhoneNumber={'@' + item.username}
        />
      </TouchableHighlight>
    );
  };

  return (
    <View style={styles.users}>
      <View style={styles.searchBox}>
        <Ionicons
          name={'arrow-back'}
          style={{paddingLeft: 10}}
          size={25}
          color={'#BFBDBD'}
          onPress={() => navigation.goBack()}
        />
        <TextInput
          style={[
            styles.search,
            {
              width: '100%',
            },
          ]}
          value={searchString}
          onChangeText={(text) => {
            setSearchString(text);
          }}
          placeholder={'Search by @username or email'}
          keyboardType={'default'}
          placeholderTextColor={'#949499'}
          autoCompleteType={'username'}
          textContentType={'username'}
          secureTextEntry={false}
        />
      </View>
      <FlatList
        ListHeaderComponent={
          <TouchableHighlight
            onPress={() => {
              if (wallet == null) {
                navigation.navigate('SelectWallet', {
                  isEditable: true,
                });
              } else {
                navigation.navigate('Send', {isEditable: true, wallet: wallet});
              }
            }}
            underlayColor="#f8f9fb">
            <SendBy
              title={'Send by address'}
              img={require('assets/img/address.png')}
            />
          </TouchableHighlight>
        }
        showsVerticalScrollIndicator={false}
        style={styles.list}
        data={users}
        renderItem={renderItem}
        keyExtractor={(item) => item.id}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  users: {
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
  },
  searchBox: {
    marginTop: 20,
    alignItems: 'center',
    flexDirection: 'row',
    width: '92%',
    backgroundColor: '#E9E9E9',
    borderRadius: 11,
  },
  search: {
    marginLeft: 10,
    padding: 0,
    fontSize: 17,
    height: 40,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: 'black',
  },
  list: {
    marginTop: 20,
    flex: 1,
    width: '100%',
  },
});
