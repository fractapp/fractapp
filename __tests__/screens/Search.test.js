import React, {useState, useContext} from 'react';
import renderer from 'react-test-renderer';
import {Search} from 'screens/Search';
import GlobalStore from 'storage/Global';
import DialogStore from 'storage/Dialog';
import {render} from '@testing-library/react-native';
import backend from 'utils/api';
import {PermissionsAndroid} from 'react-native';
import Dialog from 'storage/Dialog';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
  useContext: jest.fn(),
}));
jest.mock('storage/DB', () => ({}));
jest.mock('utils/api', () => ({
  myMatchContacts: jest.fn(),
  myContacts: jest.fn(),
  getImgUrl: jest.fn(() => 'uri'),
}));
jest.mock('react-native-contacts', () => ({
  getAll: jest.fn(),
}));

useState.mockImplementation((init) => [init, jest.fn()]);

it('Test view (empty)', () => {
  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
    dispatch: jest.fn(),
  });
  useContext.mockReturnValueOnce({
    state: DialogStore.initialState(),
    dispatch: jest.fn(),
  });

  const tree = renderer
    .create(<Search navigation={null} route={{params: {}}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test view', () => {
  const users = [];
  for (let i = 0; i < 5; i++) {
    users.push({
      id: 'id' + i,
      name: 'name' + i,
      username: 'username' + i,
      avatarExt: i % 5 === 0 ? 'png' : '',
      lastUpdate: 123123,
      addresses: {
        0: 'addressDOT' + i,
        1: 'addressKSM' + i,
      },
    });
  }

  useState
    .mockImplementationOnce((init) => [init, jest.fn()])
    .mockImplementationOnce((init) => [users, jest.fn()]);

  useContext.mockReturnValueOnce({
    state: GlobalStore.initialState(),
    dispatch: jest.fn(),
  });
  useContext.mockReturnValueOnce({
    state: DialogStore.initialState(),
    dispatch: jest.fn(),
  });

  const tree = renderer
    .create(<Search navigation={null} route={{params: {}}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
