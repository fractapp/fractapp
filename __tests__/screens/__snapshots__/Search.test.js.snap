// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view (empty) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E9E9E9",
        "borderRadius": 11,
        "flexDirection": "row",
        "marginTop": 20,
        "width": "92%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#BFBDBD",
            "fontSize": 25,
          },
          Object {
            "paddingLeft": 10,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCompleteType="username"
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="Search"
      placeholderTextColor="#949499"
      rejectResponderTermination={true}
      secureTextEntry={false}
      style={
        Array [
          Object {
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 17,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
            "marginLeft": 10,
            "padding": 0,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      textContentType="username"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <RCTScrollView
    ListHeaderComponent={
      <ForwardRef
        onPress={[Function]}
        underlayColor="#f8f9fb"
      >
        <SendBy
          img={
            Object {
              "testUri": "../../../assets/img/address.png",
            }
          }
          title="Send to address"
        />
      </ForwardRef>
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
        "marginTop": 20,
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "height": 80,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "marginLeft": "5%",
                  "marginRight": "3%",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../assets/img/address.png",
                  }
                }
                style={
                  Object {
                    "borderRadius": 30,
                    "height": 60,
                    "width": 60,
                  }
                }
              />
              <View
                style={
                  Object {
                    "marginLeft": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontFamily": "Roboto-Regukar",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                  Send to address
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Test view 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E9E9E9",
        "borderRadius": 11,
        "flexDirection": "row",
        "marginTop": 20,
        "width": "92%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#BFBDBD",
            "fontSize": 25,
          },
          Object {
            "paddingLeft": 10,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCompleteType="username"
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="Search"
      placeholderTextColor="#949499"
      rejectResponderTermination={true}
      secureTextEntry={false}
      style={
        Array [
          Object {
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 17,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
            "marginLeft": 10,
            "padding": 0,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      textContentType="username"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <RCTScrollView
    ListHeaderComponent={
      <ForwardRef
        onPress={[Function]}
        underlayColor="#f8f9fb"
      >
        <SendBy
          img={
            Object {
              "testUri": "../../../assets/img/address.png",
            }
          }
          title="Send to address"
        />
      </ForwardRef>
    }
    data={
      Array [
        Object {
          "addresses": Object {
            "0": "addressDOT0",
            "1": "addressKSM0",
          },
          "avatarExt": "png",
          "id": "id0",
          "lastUpdate": 123123,
          "name": "name0",
          "username": "username0",
        },
        Object {
          "addresses": Object {
            "0": "addressDOT1",
            "1": "addressKSM1",
          },
          "avatarExt": "",
          "id": "id1",
          "lastUpdate": 123123,
          "name": "name1",
          "username": "username1",
        },
        Object {
          "addresses": Object {
            "0": "addressDOT2",
            "1": "addressKSM2",
          },
          "avatarExt": "",
          "id": "id2",
          "lastUpdate": 123123,
          "name": "name2",
          "username": "username2",
        },
        Object {
          "addresses": Object {
            "0": "addressDOT3",
            "1": "addressKSM3",
          },
          "avatarExt": "",
          "id": "id3",
          "lastUpdate": 123123,
          "name": "name3",
          "username": "username3",
        },
        Object {
          "addresses": Object {
            "0": "addressDOT4",
            "1": "addressKSM4",
          },
          "avatarExt": "",
          "id": "id4",
          "lastUpdate": 123123,
          "name": "name4",
          "username": "username4",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
        "marginTop": 20,
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "height": 80,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "marginLeft": "5%",
                  "marginRight": "3%",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../assets/img/address.png",
                  }
                }
                style={
                  Object {
                    "borderRadius": 30,
                    "height": 60,
                    "width": 60,
                  }
                }
              />
              <View
                style={
                  Object {
                    "marginLeft": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontFamily": "Roboto-Regukar",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                  Send to address
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "height": 80,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "marginLeft": "5%",
                  "marginRight": "3%",
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "uri",
                  }
                }
                style={
                  Object {
                    "borderRadius": 30,
                    "height": 60,
                    "width": 60,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "marginLeft": 10,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "height": 23,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontFamily": "Roboto-Regukar",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    name0
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "height": 23,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#888888",
                          "fontFamily": "Roboto-Regular",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    @username0
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "height": 80,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "marginLeft": "5%",
                  "marginRight": "3%",
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "uri",
                  }
                }
                style={
                  Object {
                    "borderRadius": 30,
                    "height": 60,
                    "width": 60,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "marginLeft": 10,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "height": 23,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontFamily": "Roboto-Regukar",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    name1
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "height": 23,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#888888",
                          "fontFamily": "Roboto-Regular",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    @username1
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "height": 80,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "marginLeft": "5%",
                  "marginRight": "3%",
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "uri",
                  }
                }
                style={
                  Object {
                    "borderRadius": 30,
                    "height": 60,
                    "width": 60,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "marginLeft": 10,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "height": 23,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontFamily": "Roboto-Regukar",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    name2
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "height": 23,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#888888",
                          "fontFamily": "Roboto-Regular",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    @username2
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "height": 80,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "marginLeft": "5%",
                  "marginRight": "3%",
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "uri",
                  }
                }
                style={
                  Object {
                    "borderRadius": 30,
                    "height": 60,
                    "width": 60,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "marginLeft": 10,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "height": 23,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontFamily": "Roboto-Regukar",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    name3
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "height": 23,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#888888",
                          "fontFamily": "Roboto-Regular",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    @username3
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "height": 80,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "marginLeft": "5%",
                  "marginRight": "3%",
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "uri",
                  }
                }
                style={
                  Object {
                    "borderRadius": 30,
                    "height": 60,
                    "width": 60,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "marginLeft": 10,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "height": 23,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontFamily": "Roboto-Regukar",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    name4
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "height": 23,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#888888",
                          "fontFamily": "Roboto-Regular",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    @username4
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
